pac4j-saml/src/main/java/org/pac4j/saml/sso/impl/SAML2AuthnResponseValidator.java

```diff
@@ -195,6 +212,10 @@ protected void validateSamlProtocolResponse(final Response response, final SAML2
 
         validateSuccess(response.getStatus());
 
+        if (wantsResponseSigned && response.getSignature() == null) {
+            throw new SAMLSignatureValidationException("Signature not found");
```

Comment:
You should likely explain the root cause better here; something like:

```
Unable to find a signature on the SAML response returned. Pac4j is configured to enforce signatures on SAML2 responses from identity providers and the returned response {} does not contain a signature
```

Then log the response (by encoding it as XML first)